contentcuration/contentcuration/frontend/shared/vuex/channel/actions.js

```diff
@@ -118,10 +120,48 @@ export function commitChannel(
         channelData.content_defaults = contentDefaults;
       }
     }
+
     return Channel.createModel(channelData).then(() => {
       context.commit('UPDATE_CHANNEL', { id, ...channelData });
       context.commit('SET_CHANNEL_NOT_NEW', id);
     });
+  } else {
+    // Create a new channel
+    const newChannelData = {};
+
+    if (name !== NOVALUE) {
+      newChannelData.name = name;
+    }
+    if (description !== NOVALUE) {
+      newChannelData.description = description;
+    }
+    if (language !== NOVALUE) {
+      newChannelData.language = language;
+    }
+    if (thumbnail !== NOVALUE) {
+      newChannelData.thumbnail = thumbnail;
+    }
+    if (thumbnail_url !== NOVALUE) {
+      newChannelData.thumbnail_url = thumbnail_url;
+    }
+    if (contentDefaults !== NOVALUE) {
+      newChannelData.content_defaults = contentDefaults;
+    }
+
+    // Log the data before sending it to create a new channel
+    console.log('Creating new channel with data:', newChannelData);
+
+    return Channel.createModel(newChannelData).then(response => {
+      console.log('API Response:', response);
+
+      const createdChannel = response.data || response; // Adjust based on API structure
```

Comment:
It must be one or the other-- either `response.data` or `response`? I suggest making this specific.